pkg/resource/v1/chartconfig/desired_test.go

```diff
@@ -0,0 +1,101 @@
+package chartconfig
+
+import (
+	"context"
+	"reflect"
+	"testing"
+
+	"github.com/giantswarm/apiextensions/pkg/apis/core/v1alpha1"
+	"github.com/giantswarm/apiextensions/pkg/clientset/versioned/fake"
+	"github.com/giantswarm/micrologger/microloggertest"
+	apimetav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clientgofake "k8s.io/client-go/kubernetes/fake"
+
+	"github.com/giantswarm/cluster-operator/pkg/cluster"
+)
+
+func Test_GetDesiredState(t *testing.T) {
+	testCases := []struct {
+		name           string
+		obj            interface{}
+		expectedLabels map[string]string
+		expectedSpec   v1alpha1.ChartConfigSpec
+		expectedType   apimetav1.TypeMeta
+	}{
+		{
+			name: "basic match",
+			obj: &v1alpha1.ClusterGuestConfig{
+				DNSZone: "5xchu.aws.giantswarm.io",
+				ID:      "5xchu",
+				Owner:   "giantswarm",
+			},
+			expectedLabels: map[string]string{
+				"giantswarm.io/cluster":      "5xchu",
+				"giantswarm.io/managed-by":   "cluster-operator",
+				"giantswarm.io/organization": "giantswarm",
+			},
+			expectedSpec: v1alpha1.ChartConfigSpec{
+				Chart: v1alpha1.ChartConfigSpecChart{
+					Name:    "quay.io/giantswarm/chart-operator-chart",
+					Channel: "stable",
+					Release: "chart-operator",
+				},
+				VersionBundle: v1alpha1.ChartConfigSpecVersionBundle{
+					Version: "0.1.0",
+				},
+			},
+			expectedType: apimetav1.TypeMeta{
+				Kind:       "ChartConfig",
+				APIVersion: "core.giantswarm.io",
+			},
+		},
+	}
+
+	var err error
+	var newResource *Resource
```

Comment:
It is not good to define "global" variables which are manipulated multiple times within a loop. Lets please have these here inside the loop as well to prevent nasty side effects and run each test with its initialized fresh values from scratch. I am seeing this also in the aws-operator a lot.